hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/aggregate/JsonSqlAggregateTest.java

```diff
@@ -185,4 +229,71 @@ private static String createTable() {
         );
         return name;
     }
+
+    private static String createTable2() {
+        String name = randomName();
+        TestBatchSqlConnector.create(
+                sqlService,
+                name,
+                asList("name", "k", "v"),
+                asList(VARCHAR, VARCHAR, VARCHAR),
+                asList(new String[]{"Alice", "department", "dep1"},
+                        new String[]{"Bob", "department", "dep2"},
+                        new String[]{"Alice", "job", "job1"},
+                        new String[]{null, "department", "dep2"},
+                        new String[]{"Bob", "job", "job2"},
+                        new String[]{null, "job", "job1"},
+                        new String[]{null, null, "desc2"},
+                        new String[]{"Bob", "description", null},
+                        new String[]{"Alice", "description", "desc1"})
+        );
+        return name;
+    }
+
+    private void assertRowsWithJsonValues(String sql, Collection<Row> rows) {
+        Map<Object, JsonValue> rowsInMap = new HashMap<>();
+        for (Row row : rows) {
+            Object[] rowObj = row.getValues();
+            if (rowObj.length != 2) {
+                throw new AssertionError("Row length must be 2");
+            }
+            if (!(rowObj[1] instanceof HazelcastJsonValue)) {
+                throw new AssertionError("Second element in the row must be HazelcastJsonValue");
+            }
+            JsonValue value = Json.parse(((HazelcastJsonValue) rowObj[1]).getValue());
+            rowsInMap.put(rowObj[0], value);
+        }
+
+        SqlResult result = sqlService.execute(sql);
+        Map<String, JsonValue> actualRowsInMap = new HashMap<>();
+        result.iterator().forEachRemaining(r -> {
+            if (r.getMetadata().getColumnCount() != 2) {
+                throw new AssertionError("The length of the result row is not 2");
+            }
+            JsonValue value = Json.parse(((HazelcastJsonValue) r.getObject(1)).getValue());
+            actualRowsInMap.put(r.getObject(0), value);
+        });
+
+        if (rowsInMap.size() != actualRowsInMap.size()) {
+            throw new AssertionError("Number of expected rows is different than actual");
+        }
+        for (Map.Entry<Object, JsonValue> kv : rowsInMap.entrySet()) {
+            JsonValue value = actualRowsInMap.get(kv.getKey());
+            JsonObject object = value.asObject();
+            JsonObject object2 = kv.getValue().asObject();
+            if (!(jsonObjectEquals(object, object2))) {
+                throw new AssertionError("Object: " + object + " is not equal to Object: " + object2);
+            }
+        }
+    }
+
+    private boolean jsonObjectEquals(JsonObject obj, JsonObject obj2) {
```

Comment:
If I read correctly that method allows mixed order of attributes only on the first level, so sucha a JSONs are not equal:
```json
{
    "foo": {
        "bar1" : "bar1",
        "bar2" : "bar2"
    }
}
```
```json
{
    "foo": {
        "bar2" : "bar2",
        "bar1" : "bar1"
    }
}
```

If I'm correct then maybe you should rename the method so it inform about that. But it is just a minor.